src/main/java/net/atos/zac/authentication/UserPrincipalFilter.java

```diff
@@ -22,40 +25,56 @@
 import org.wildfly.security.http.oidc.OidcPrincipal;
 import org.wildfly.security.http.oidc.OidcSecurityContext;
 
+import net.atos.zac.zaaksturing.ZaakafhandelParameterService;
+import net.atos.zac.zaaksturing.model.ZaakafhandelParameters;
+
 @WebFilter(filterName = "UserPrincipalFilter")
 public class UserPrincipalFilter implements Filter {
 
     private static final Logger LOG = Logger.getLogger(UserPrincipalFilter.class.getName());
 
-    private static final String GROUP_MEMBERSHIP_CLAIM_NAME = "group_membership";
+    private static final String ROL_DOMEIN_ELK_ZAAKTYPE = "domein_elk_zaaktype";
+
+    @Inject
+    private ZaakafhandelParameterService zaakafhandelParameterService;
 
     @Override
     public void init(final FilterConfig filterConfig) throws ServletException {
         Filter.super.init(filterConfig);
     }
 
     @Override
-    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain)
-            throws ServletException, IOException {
+    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse,
+            final FilterChain filterChain) throws ServletException, IOException {
         if (servletRequest instanceof HttpServletRequest) {
             final HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
             final OidcPrincipal principal = (OidcPrincipal) httpServletRequest.getUserPrincipal();
-            HttpSession httpSession = httpServletRequest.getSession(true);
 
             if (principal != null) {
+                HttpSession httpSession = httpServletRequest.getSession(true);
                 LoggedInUser loggedInUser = SecurityUtil.getLoggedInUser(httpSession);
                 if (loggedInUser != null && !loggedInUser.getId().equals(principal.getName())) {
-                    LOG.info(String.format("HTTP session of user '%s' on context path %s is invalidated", loggedInUser.getId(),
-                                           httpServletRequest.getServletContext().getContextPath()));
-                    loggedInUser = null;
-                    httpSession.invalidate();
-                    httpSession = httpServletRequest.getSession(true);
+                    synchronized (this) {
```

Comment:
Je synchronized op this, dus de instance van het filter. Ik weet eerlijk gezegd niet hoeveel instances er kunnen zijn. Misschien moet je synchronize op de class dan weet je zeker dat je het op een singleton doet. Of misschien beter nog op de session zelf, want dat is het object waar je die laze variable 1x op wilt zetten.